cmds/core/sluinit/uinit.go

```diff
@@ -73,13 +73,35 @@ func main() {
 	}
 	slaunch.Debug("Collectors completed")
 
-	slaunch.Debug("********Step 4: Write eventlog to /boot partition*********")
-	if e := p.EventLog.Persist(); e != nil {
-		log.Printf("EventLog.Persist() failed err=%v", e)
-		os.Exit(1)
+	slaunch.Debug("********Step 4: Measuring target kernel, initrd ********")
+	if e := p.Launcher.MeasureKernel(tpmDev); e != nil {
+		log.Printf("Launcher.MeasureKernel failed err=%v", e)
+		unmountAndExit()
 	}
 
-	slaunch.Debug("********Step 5: Launcher called ********")
-	err = p.Launcher.Boot(tpmDev)
-	log.Printf("Boot failed. err=%s", err)
+	slaunch.Debug("********Step 5: Parse eventlogs *********")
+	if e := p.EventLog.Parse(); e != nil {
+		log.Printf("EventLog.Parse() failed err=%v", e)
+		unmountAndExit()
+	}
+
+	slaunch.Debug("*****Step 6: Dump logs to disk *******")
+	if e := slaunch.ClearPersistQueue(); e != nil {
+		log.Printf("ClearPersistQueue failed err=%v", e)
+		unmountAndExit()
+	}
+
+	slaunch.Debug("********Step *: Unmount all ********")
+	slaunch.UnmountAll()
+
+	slaunch.Debug("********Step 7: Launcher called to Boot ********")
+	if err := p.Launcher.Boot(tpmDev); err != nil {
+		log.Printf("Boot failed. err=%s", err)
+		unmountAndExit()
+	}
+}
```

Comment:
If you follow the defer recommendation, the last call should be `os.Exit(0)` to short out the defer to `unmountAndExit`